up42/storage.py

```diff
@@ -204,47 +219,70 @@ def get_assets(
 
     def get_orders(
         self,
+        workspace_orders: bool = True,
         return_json: bool = False,
         limit: Optional[int] = None,
         sortby: str = "createdAt",
         descending: bool = True,
         order_type: Optional[str] = None,
+        statuses: Optional[List[AllowedStatuses]] = None,
+        name: Optional[str] = None,
         tags: Optional[List[str]] = None,
     ) -> Union[List[Order], dict]:
         """
-        Gets all orders in the workspace as Order objects or JSON.
+        Gets all orders in the account/workspace as Order objects or JSON.
 
         Args:
+            workspace_orders: If set to True, only returns workspace orders. Otherwise, returns all account orders.
             return_json: If set to True, returns JSON object.
             limit: Optional, only return n first assets by sorting criteria and order.
                 Optimal to select if your workspace contains many assets.
             sortby: The sorting criteria, one of "createdAt", "updatedAt", "status", "dataProvider", "type".
             descending: The sorting order, True for descending (default), False for ascending.
             order_type: Can be either "TASKING" or "ARCHIVE". Pass this param to filter orders based on order_type.
+            statuses: Search for orders with any of the statuses provided.
+            name: Search for orders that contain this string in their name.
             tags: Search for orders with any of the provided tags.
 
         Returns:
             Order objects in the workspace or alternatively JSON info of the orders.
         """
-        allowed_sorting_criteria = [
+        allowed_statuses = {
+            "CREATED",
+            "BEING_PLACED",
+            "PLACED",
+            "PLACEMENT_FAILED",
+            "DELIVERY_INITIALIZATION_FAILED",
+            "BEING_FULFILLED",
+            "DOWNLOAD_FAILED",
+            "DOWNLOADED",
+            "FULFILLED",
+            "FAILED_PERMANENTLY",
+        }
+
+        allowed_sorting_criteria = {
             "createdAt",
             "updatedAt",
             "type",
             "status",
-            "dataProvider",
-        ]
+        }
         if sortby not in allowed_sorting_criteria:
```

Comment:
We must have here the same approach as with statuses, but this is for another task

Reply:
Yes, also please note the approach for handling url query params changed on the order method for `storage.py`, this approach should be followed in all the other methods across the SDK, but this is out of the scope. I will try to add it to each class refactoring.

Reply:
statuses are not blocking the method, but instead ignore the values that are not allowed. That was the way it was intended from the start. I have added a log to inform the user about which values were ignored. Open to change as well.